lib/ansible/plugins/inventory/__init__.py

```diff
@@ -318,6 +316,7 @@ def _add_host_to_composed_groups(self, groups, variables, host, strict=False):
         if groups and isinstance(groups, dict):
             self.templar.set_available_variables(variables)
             for group_name in groups:
+                group_name = to_safe_group_name(group_name)
```

Comment:
If the group name changes here the next line `groups[group_name]` is a KeyError.

Reply:
he, missed that